lntemp/harness_node_manager.go

```diff
@@ -0,0 +1,210 @@
+package lntemp
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntemp/node"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+)
+
+// nodeManager is responsible for hanlding the start and stop of a given node.
+// It also keeps track of the running nodes.
+type nodeManager struct {
+	sync.Mutex
+
+	// chainBackend houses the information necessary to use a node as LND
+	// chain backend, such as rpc configuration, P2P information etc.
+	chainBackend node.BackendConfig
+
+	// currentTestCase holds the name for the currently run test case.
+	currentTestCase string
+
+	// lndBinary is the full path to the lnd binary that was specifically
+	// compiled with all required itest flags.
+	lndBinary string
```

Comment:
Does this mean we won't attempt to run tests in parallel any longer? Or we'll instead map `M` tests onto `N` node managers which also execute the tests (so not using the current `xargs` ting).

Reply:
I think the parallel part is unchanged.